### PR TITLE
ARM builders

### PR DESCRIPTION
**Pull request by albertvaka**

### CODE REVIEW

File: deb-arm64/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+RUN curl -sL -O https://dd-agent-omnibus.s3.amazonaws.com/cmake-${CMAKE_VERSION}-ubuntu-aarch64.tar.xz && \
+    tar xf cmake-${CMAKE_VERSION}-ubuntu-amd64.tar.xz --no-same-owner -kC / && \
+    rm cmake-${CMAKE_VERSION}-ubuntu-amd64.tar.xz
+ENV PATH="/opt/cmake/bin:$PATH"
+
+# Install clang and llvm version 8. Pre-built because building takes ~4 hours.
+# This was built from sources on centos 7, using the build-clang.sh script
+RUN curl -sL -O https://dd-agent-omnibus.s3.amazonaws.com/clang%2Bllvm-${CLANG_VERSION}-aarch64-linux.tar.xz && \
+    tar xf clang+llvm-${CLANG_VERSION}-amd64-linux.tar.xz --no-same-owner -kC / && \
+    rm clang+llvm-${CLANG_VERSION}-amd64-linux.tar.xz
```
Comment (cricket007) [inline deb-arm64/Dockerfile]: `s/amd64/aarch64`?

File: deb-arm64/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# Gimme
+RUN curl -sL -o /bin/gimme https://raw.githubusercontent.com/travis-ci/gimme/master/gimme
+RUN chmod +x /bin/gimme
+RUN gimme $GIMME_GO_VERSION
+COPY ./gobin.sh /etc/profile.d/
+
+# CMake. Pre-built using the build-cmake.sh script, to speed-up docker build.
+RUN curl -sL -O https://dd-agent-omnibus.s3.amazonaws.com/cmake-${CMAKE_VERSION}-ubuntu-aarch64.tar.xz && \
+    tar xf cmake-${CMAKE_VERSION}-ubuntu-amd64.tar.xz --no-same-owner -kC / && \
+    rm cmake-${CMAKE_VERSION}-ubuntu-amd64.tar.xz
```
Comment (cricket007) [inline deb-arm64/Dockerfile]: `s/amd64/aarch64`?

File: rpm-arm64/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+# Gimme
+RUN curl -sL -o /bin/gimme https://raw.githubusercontent.com/travis-ci/gimme/master/gimme
+RUN chmod +x /bin/gimme
+RUN gimme $GIMME_GO_VERSION
+COPY ./gobin.sh /etc/profile.d/
+
+# CMake. Pre-built using the build-cmake.sh script, to speed-up docker build.
+RUN curl -sL -O https://dd-agent-omnibus.s3.amazonaws.com/cmake-${CMAKE_VERSION}-amzn-aarch64.tar.xz && \
+    tar xf cmake-${CMAKE_VERSION}-amzn-arm64.tar.xz --no-same-owner -kC / && \
+    rm cmake-${CMAKE_VERSION}-amzn-arm64.tar.xz
```
Comment (cricket007) [inline rpm-arm64/Dockerfile]: `s/arm64/aarch64`?

File: rpm-arm64/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ENV PATH="/opt/cmake/bin:$PATH"
+
+# Install clang and llvm version 8. Pre-built because building takes ~4 hours.
+# This was built from sources on centos 7, using the build-clang.sh script
+RUN curl -sL -O https://dd-agent-omnibus.s3.amazonaws.com/clang%2Bllvm-${CLANG_VERSION}-aarch64-linux.tar.xz && \
+    tar xf clang+llvm-${CLANG_VERSION}-arm64-linux.tar.xz --no-same-owner -kC / && \
+    rm clang+llvm-${CLANG_VERSION}-arm64-linux.tar.xz
```
Comment (cricket007) [inline rpm-arm64/Dockerfile]: `s/arm64/aarch64`?

**Comment by albertvaka** [comment]
Thanks for your interest! We have been working on this but didn't update the PR. Done that now :)

**Comment by cricket007** [comment]
Thanks for the update @albertvaka 

I was able to edit myself to get the container to build, however now the omnibus-software package that gets downloaded seems to have an old cacerts sha256 sum. 

Would that need to be addressed in a different repo?

**Comment by KSerrania** [comment]
Hi @cricket007, the cacerts error is happening because the cacerts bundle has changed this morning (see here: https://curl.haxx.se/docs/caextract.html). We'll update the `omnibus-software` repository to match this update.

**Comment by cricket007** [comment]
Ah, that would explain it! 

Following all these updates, would I still need the container to build the packages? Or is the end result that there will be arm64 DEB + RPM packages available to download?

**Comment by albertvaka** [comment]
DEB and RPM packages are "coming soon" :) If waiting a bit is an option you won't have to go through the hassle of the build.

**Comment by cricket007** [comment]
It's just for a homelab; nothing critical. I certainly can wait 😎

**Comment by albertvaka** [comment]
APT: 
```
https://apt.datadoghq.com beta 7
```
RPM:
```
[datadog]
name = Datadog, Inc.
baseurl = https://yum.datadoghq.com/beta/7/aarch64/
enabled=1
gpgcheck=1
repo_gpgcheck=0
priority=1
gpgkey=https://yum.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public
```

**Pull request merged**